### ISSUE TITLE
@append/prependEdge behavior when node already exists

### ISSUE THREAD

**Issue by tsnobip** [facebook/relay#3838]
Today, since https://github.com/facebook/relay/pull/3373 got merged, when using `@append/prependEdge` on an already existing node, no node is appended/prepended. 

I get that the previous behavior (ie adding a duplicate node) was broken, but I still think that the current behavior is not ideal, shouldn't we instead move the existing node to bottom/top instead of doing nothing?

If you think the current behavior is still sometimes wanted, we could add an additional parameter like:
```
@prependEdge(whenExists: false)
```

Would you be open to a PR that corrects this behavior?

**Comment by voideanvalue** [comment]
Conceptually, a node appearing multiple times in a connection seems odd to me. What are your use cases where that's desirable?

That being said... I believe it is still possible to achieve this behavior with [updater functions](https://relay.dev/docs/guided-tour/updating-data/graphql-mutations/#updater-functions). So unless there's evidence to support that this behavior would find frequent usage, I don't feel inclined to add something like `whenExists`.

**Comment by tsnobip** [comment]
I agree, the node should not appear multiple times in a connection, that was the previous behavior that got corrected with https://github.com/facebook/relay/pull/3373 by not doing anything when this happens, but I think it would be more coherent to remove the existing node and add the updated node to the top/bottom of the list as the directive states.

The use case is quite common I think, it's when you want to upsert an entity. If the entity doesn't exist yet, it will be prepended/appended as expected, if it already exists it will not be prepended appended, which can be weird especially when the connection is sorted by `last_requested` for example. 

I see no reason why this behavior would find less frequent usage than the current one (not moving the node), and in case we would like to support the two behavior (if both would be roughly as common) we could add the additional parameter I suggested.

**Comment by tsnobip** [comment]
@voideanvalue any thoughts on my comment?
